Update security.md (#293)
subspace/subspace-docs: docs/protocol/security.md

---
title: General and server security guidelines
sidebar_position: 5
description: General security measures. Cyberattacks and fraud protection. Server protection.
keywords:
  - Security
  - Wallet
  - Password
  - Server
  - Protection
  - Fraud
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import Link from '@docusaurus/Link';
import styles from '@site/src/pages/index.module.css';

Navigating the internet and being part of a crypto project can be exciting! But it's crucial to use best security practices. Here is a general guide to security and safety advice, including basic measures for securing servers and home PCs. This is your contribution to keeping Subspace network up and running and of course the safety of the community's cryptoassets. 

Enjoy Subspace journey with peace of mind!

> # 1. Wallet security

Author: Hmm. Yes, really without paragraph numbers, the whole document is easier to read! Thank you

### Passwords
*

>  Be sure to make the password long enough, combining both registers letters, numbers and special characters. An 11-character password increases the bruteforce attack time to normally impossible limit but plese take a look on [passwords brutforce time illustrating table:](https://www.techrepublic.com/wp-content/uploads/2022/03/tr8423-Figure-A-How-an-8-character-password-could-be-cracked-in-just-a-few-minutes-770x404.jpg).

Author: Thank you :)

*
 Be cautious of public Wi-Fi
*
 Keep software up to date
*
 If possible, enable 2FA (two-factor authentication) for all your cryptocurrency accounts and wallets. This provides an extra layer of security by requiring a second verification step, such as a code from a mobile app or hardware token, in addition to a password.
*
 Secure your devices! And here's a little basic guide for servers and home PCs in section #3.

# 2. Community security
*
 Remember: Subspace team members will never send you direct messages first. If this happens to you, ignore the message, block the sender, and report the incident to our ["scam report" channel](https://discord.com/channels/864285291518361610/982023010405990400)
*
 Be ware of fishing attacks.
Do not click on any links on our forum, Discord, or Telegram unless these links are posted by one of team members (ambassadors or staff). Scammers may create fake links looking like legit websites, so the easiest way to stay safe is to avoid such messages at all.

# 3. Basic server securing

> *let's assume you just got a new server*

Reviewer: I'd suggest to add a note about why it's needed, instead of this message and `etc ... check every SSD`

### check SSD' condition (while in a rescue system):
```console
    apt-get -y install nvme-cli
    nvme list       
    nvme --smart-log /dev/ng0n1
    nvme --smart-log /dev/ng1n1
```
etc ... check every SSD

### **SSH port change**
*

> First make sure you have not UFW enabled. Or add the rule with the desired port to it.

Reviewer: ```suggestion
First, make sure you don't have UFW enabled. If you do, add a rule for the desired port."
```

```console
sudo vi /etc/ssh/sshd_config
#Port 22  -> Port 12345             # specify a custom SSH port within the range of 1025 to 65534
sudo systemctl reload sshd    # for the changes to take effect
```

>  *Remember: add this port to UFW*

Reviewer: We can use `:::note` to communicate this better

```console
sudo ufw allow xxxxx/tcp
```
*
Checking new port from homePC:
```console
ssh -p xxxxx user@localhost
```

### SSH key secure login
*
On the server:
```console
vi /etc/ssh/sshd_config
--> PubkeyAuthentication yes
systemctl restart ssh
```
*
Create rsa keys on home PC:

if you don't have keys.
```console
ssh-keygen
```
you will get two keys:

`$HOME/.ssh/id_rsa – Your private RSA key to keep on local PC`

`$HOME/.ssh/id_rsa.pub – Your public RSA key to send to a server`
*
Transfer rsa keys to the server:

On a home PC:

```console
sudo ssh-copy-id -p xxxxx user_name@server-ip-addres
``` 
This utility specifically designed for copying SSH keys to a remote server. 
It automatically handles the key placement and permissions on the remote server, making it more convenient. 
Use -p flag to specify not standard port

if you have created keys before and keep them in somewhere else, use rsync:
```console
sudo rsync -e "ssh -p xxxxx" ~/.ssh/id_rsa.pub USER@SERVER_IP:~/.ssh/authorized_keys
```		

> manually transfer the public key file and add it to the authorized_keys file on the remote server. 

Reviewer: ```suggestion
Manually transfer the public key file and add it to the authorized_keys file on the remote server. 
```

Author: Got it

To view the changelog for a particular package: 

`apt changelog <package_name>`
*
Upgrading Kernel.
While kernel updates often come with bug fixes and security patches, there is a possibility that the new kernel version may introduce new bugs or compatibility issues. Not every kernel update is necessary or urgent. Some updates may provide incremental improvements or additional functionality that may not be essential for your specific use case. It's important to evaluate the benefits and potential risks before deciding to update the kernel.
*
Upgrading the distribution version have both pros and cons.
Pros:
    Access to new features
    Software compatibility
    Security updates
    Long-term support (LTS)
Cons:
    Potential for compatibility issues
    Configuration changes
    May have new bugs (can be resolved by downgrading the bugged package)

> To get in touch of Release Notes simply use search https://ubuntu.com/search?q=Ubuntu+22.04+Changelogs

Reviewer: Let's make this a hyperlink
